### PR TITLE
Cmake super project

### PR DESCRIPTION
**Pull request by ethanrublee**
Inspiration from https://github.com/cryos/openchemistry and https://www.kitware.com/cmake-superbuilds-git-submodules/

also see: https://github.com/farm-ng/farm-ng-core/pull/14

### CODE REVIEW

File: scripts/example_super_project/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# Place this file in a directory with:
+# - farm-ng-core # branch cmake-super-project
```
Comment (ethanrublee) [inline scripts/example_super_project/CMakeLists.txt:2]: @strasdat this can be done at CI level - handles installing most of the dependencies from source builds.

**Comment by strasdat** [comment]
@ethanrublee - CI is not triggered. Let me see why...

File: cpp/sophus/image/CMakeLists.txt [hunk 2]
```diff
@@ -21,23 +21,23 @@ foreach(x ${sophus_image_src_prefixes} )
...
-               BASE_DIRS ../..
-               FILES ${sophus_image_h})
 
 
 if(${BUILD_SOPHUS_TESTS})
   foreach(test_basename ${sophus_image_src_prefixes})
     farm_ng_add_test(${test_basename}
                           PARENT_LIBRARY sophus_image
-                          LINK_LIBRARIES
+                          LINK_LIBRARIES sophus_image
```
Comment (strasdat) [inline cpp/sophus/image/CMakeLists.txt:40]: why is that needed? should'nt we always link against the parent lib?

Comment (ethanrublee) [reply cpp/sophus/image/CMakeLists.txt:40]: For some reason the macro didn't link against the parent lib, which is used just for the test name. We can make it default in a follow up.

File: CMakeLists.txt [hunk 3]
```diff
@@ -59,7 +59,6 @@ endif()
 
 if (BUILD_FARM_NG_PROTOS)
   find_package(Protobuf REQUIRED)
-  include_directories(${Protobuf_INCLUDE_DIRS})
```
Comment (strasdat) [inline CMakeLists.txt:61]: good catch!

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -28,16 +28,6 @@ jobs:
             access: row_major
```
Comment (ethanrublee) [inline .github/workflows/main.yml:28]: @strasdat please note that row_major / col_major is broken, as Eigen is installed as part of the super_project_build, and this needs to be plumbed into that project.

Comment (strasdat) [reply .github/workflows/main.yml:28]: It seems like the env variable is set correctly here. Let me check...

Comment (strasdat) [reply .github/workflows/main.yml:28]: okay, this is not set correctly in the main.yml. Turning it off for now...

Comment (strasdat) [reply .github/workflows/main.yml]: @ethanrublee, I made a merge mistake. Will re-enable your changes... about row-major.
